model/AbstractRdsResultStorage.php

```diff
@@ -0,0 +1,727 @@
+
+        if (isset($options['offset'])) {
+            $qb->setFirstResult($options['offset']);
+        }
+        if (isset($options['limit'])) {
+            $qb->setMaxResults($options['limit']);
+        }
+
+        if (count($delivery) > 0) {
+            $qb
+                ->andWhere(self::DELIVERY_COLUMN . ' IN (:delivery)')
+                ->setParameter(':delivery', $delivery, Connection::PARAM_STR_ARRAY);
+        }
+
+        $returnValue = [];
+        foreach ($qb->execute()->fetchAll() as $value) {
+            $returnValue[] = [
+                self::FIELD_DELIVERY_RESULT => $value[self::RESULTS_TABLE_ID],
+                self::FIELD_TEST_TAKER => $value[self::TEST_TAKER_COLUMN],
+                self::FIELD_DELIVERY => $value[self::DELIVERY_COLUMN],
+            ];
+        }
+
+        return $returnValue;
+    }
+
+    /**
+     * Generates and sanitize ORDER BY field.
+     *
+     * @param array $options
+     *
+     * @return string
+     */
+    protected function getOrderField(array $options)
+    {
+        $allowedOrderFields = [self::DELIVERY_COLUMN, self::TEST_TAKER_COLUMN, self::RESULTS_TABLE_ID];
+
+        if (isset($options['order']) && in_array($options['order'], $allowedOrderFields)) {
+            return $options['order'];
+        }
+
+        return self::RESULTS_TABLE_ID;
+    }
+
+    /**
+     * Generates and sanitize ORDER BY direction.
+     *
+     * @param array $options
+     *
+     * @return string
+     */
+    protected function getOrderDirection(array $options)
+    {
+        $allowedOrderDirections = ['ASC', 'DESC'];
+
+        if (isset($options['orderdir']) && in_array(strtoupper($options['orderdir']), $allowedOrderDirections)) {
+            return $options['orderdir'];
+        }
+
+        return 'ASC';
+    }
+
+    public function countResultByDelivery($delivery)
+    {
+        if (!is_array($delivery)) {
+            $delivery = [$delivery];
+        }
+        $qb = $this->getQueryBuilder()
+            ->select('COUNT(*)')
+            ->from(self::RESULTS_TABLENAME);
+
+        if (count($delivery) > 0) {
+            $qb
+                ->andWhere(self::DELIVERY_COLUMN . ' IN (:delivery)')
+                ->setParameter('delivery', $delivery, Connection::PARAM_STR_ARRAY);
+        }
+
+        return $qb->execute()->fetchColumn();
+    }
+
+    public function deleteResult($deliveryResultIdentifier)
+    {
+        // remove variables
+        $sql = 'DELETE FROM ' . self::VARIABLES_TABLENAME . '
+            WHERE ' . self::VARIABLES_FK_COLUMN . ' = ?';
+
+        if ($this->getPersistence()->exec($sql, [$deliveryResultIdentifier]) === false) {
+            return false;
+        }
+
+        // remove results
+        $sql = 'DELETE FROM ' . self::RESULTS_TABLENAME . '
+            WHERE ' . self::RESULTS_TABLE_ID . ' = ?';
+
+        if ($this->getPersistence()->exec($sql, [$deliveryResultIdentifier]) === false) {
```

Comment:
use return instead maybe? wdyt ?

Reply:
Directly return would end the method execution here, even in case of `true` return value.